Vertexvis/vertex-web-sdk
# [PLAT-2134] Add support for WebSocketTransport-based Scene Tree subscriptions

## Summary

Adds a new flag to the scene-tree controller's `ConnectOptions` that enables usage of the `WebsocketTransport` for subscription requests (defaulted to `true`). This will then use `grpc.setDefaultTransport` to modify the default transport prior to the subscribe request, and immediately set it back to the default `CrossBrowserHttpTransport`. This is supported on a per-request basis as well, but due to the way our protobuf packages are generated, we don't have the ability to modify the transport for those directly.

## Test Plan

- Verify that scene tree requests work as expected (expand/collapse/etc)
- Verify that scene tree subscriptions are over websocket by default
- Verify that scene tree subscriptions are over http when `subscriptionUseWebSocketTransport` is false

## Release Notes

N/A

## Possible Regressions

Scene tree interactions/subscription

## Dependencies

https://github.com/Vertexvis/scene-tree-service/pull/120
